public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -163,6 +163,30 @@ export class LokiDatasource
       ...options.scopedVars,
       ...this.getRangeScopedVars(options.range),
     };
+
+    // if all these are true, run query through backend:
+    // - feature-flag is enabled
+    // - we are in explore-mode
+    // - for every query it is true that:
+    //   - query is range query
+    //   - and query is metric query
+    //   - and query is not a log-volume-query (those need a custom http header)
+    const shouldRunBackendQuery =
+      config.featureToggles.lokiBackendMode &&
+      options.app === CoreApp.Explore &&
+      options.targets.every(
+        (query) => query.queryType === LokiQueryType.Range && isMetricsQuery(query.expr) && !query.volumeQuery
+      );
+
+    if (shouldRunBackendQuery) {
+      // we "fix" the loki queries to have `.queryType` and not have `.instant` and `.range`
+      const fixedOptions = {
```

Comment:
This is not related to your PR, but do you think it would make sense to rename `options` variable to request as its type is `DataQueryRequest<LokiQuery>` and it is basically a request. It makes it more clear for me. What do you think?

Reply:
sure, done!